File: backend/src/main/java/ru/vsu/cs/sportbox/Data/Model/Event.java
package ru.vsu.cs.sportbox.Data.Model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "event")
public class Event {
    @Id
    @Column(name = "id", unique = true)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name")
    private String name;

    @Column(name = "price")
    private Double price;

    @Temporal(TemporalType.DATE)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @Column(name = "start_date")
    private Date startDate;

    @Temporal(TemporalType.DATE)
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @Column(name = "end_date")
    private Date endDate;

    @Column(name = "description")
    private String description;

    @ManyToOne (optional=false, cascade=CascadeType.ALL)
    @JoinColumn (name="inventory_type_id")
    @JsonIgnore
    private InventoryType inventoryType;

    @ManyToMany(mappedBy="events", fetch = FetchType.EAGER)
    @JsonIgnore
    private List<Booking> bookings;
}

File: frontend/src/components/Event.js
import React from "react";

function Event({ event }) {
    return (
        <div className="event-wrapper">
            <div className="event-img">
                <img src={require("../images/events/" + event.name + ".jpg") } alt="MISSING JPG"></img>
            </div>
            <div className="event-info-wrapper">
                <div className="event-name">
                    {event.name}
                </div>

                <div className="event-description">
                    {event.description}
                </div>

                <div className="event-equipment">
                    Необходимое оборудование: {event.inventoryType.type}
                </div>

                <div className="event-date">
                    Даты проведения: с {event.startDate} по {event.endDate}
                </div>

                <div className="event-price">
                    {event.price} руб.
                </div>
            </div>
        </div>
    );
};

export default Event;